src/Gallery/index.js

```diff
@@ -77,10 +77,42 @@ const TopInfo = styled.div`
   flex-direction: row;
   justify-content: center;
 `;
-export default ({ nfts }) => {
-  const [selectedCategory, selectCategory] = useState("All");
-  const [selectedNetwork, selectNetwork] = useState();
-  const [sortBy, setSortBy] = useState();
+
+
+const sort = (nfts, sortBy) => {
+  // console.log(nfts)
+  console.log(sortBy)
+  const forsale = nfts.filter((nft) => (nft.forSale === true))
+  const notforsale = nfts.filter((nft) => (nft.forSale === false))
+  let compareFn;
+  switch (sortBy) {
+    case 'lowTohigh':
+      compareFn = (a, b) => a.price - b.price
+      break;
+      case 'highToLow':
+        compareFn = (a, b) => ( b.price - a.price)
+        break;
+      case 'rarityAscending':
+        compareFn = (a, b) => a.rarity - b.rarity
+        break;
+      case 'rarityDescending':
+        compareFn = (a,b) => b.rarity - a.rarity
+      }
+      let sorted =  forsale.sort(compareFn)
+      sorted.push(...notforsale)
+      return sorted
+    }
```

Comment:
Only price sorting should put "not for sale" at the back. Rarity sorting should disregard "for sale" status.